# Model Evaluation
## Compare the General Model's Performance on Boston Celtics specifically vs. Model Trained on Boston Celtics Data Alone

In [4]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt

In [13]:
bos_general = pd.read_csv('gen_model_boston_celtics_eval.csv')
bos_specific = pd.read_csv('boston_celtics_model_eval.csv')

# Calculate the difference between Actual PTS and Predicted PTS
bos_general['Point Diff'] = bos_general['Actual PTS'] - bos_general['Predicted PTS']
bos_specific['Point Diff'] = bos_specific['Actual PTS'] - bos_specific['Predicted PTS']

# Descriptive Statistics for Point Differences
gen_mean = bos_general['Point Diff'].mean()
gen_median = bos_general['Point Diff'].median()
gen_std = bos_general['Point Diff'].std()
gen_min = bos_general['Point Diff'].min()
gen_max = bos_general['Point Diff'].max()

bos_mean = bos_specific['Point Diff'].mean()
bos_median = bos_specific['Point Diff'].median()
bos_std = bos_specific['Point Diff'].std()
bos_min = bos_specific['Point Diff'].min()
bos_max = bos_specific['Point Diff'].max()

print("--- General Model Descriptive Statistics ---")
print(f"Mean: {gen_mean:.2f}, Median: {gen_median:.2f}, Std Dev: {gen_std:.2f}, Range: ({gen_min:.2f}, {gen_max:.2f})")
print("\n--- Celtics-Specific Model Descriptive Statistics ---")
print(f"Mean: {bos_mean:.2f}, Median: {bos_median:.2f}, Std Dev: {bos_std:.2f}, Range: ({bos_min:.2f}, {bos_max:.2f})")

# Unpaired t-test
t_stat, p_value = stats.ttest_ind(bos_general['Point Diff'], bos_specific['Point Diff'], equal_var=False)

print("\n--- Unpaired t-test ---")
print(f"t-statistic: {t_stat:.2f}, p-value: {p_value:.4f}")


--- General Model Descriptive Statistics ---
Mean: 2.24, Median: 1.74, Std Dev: 9.30, Range: (-30.10, 43.08)

--- Celtics-Specific Model Descriptive Statistics ---
Mean: 12.53, Median: 11.17, Std Dev: 23.77, Range: (-53.60, 62.11)

--- Unpaired t-test ---
t-statistic: -3.45, p-value: 0.0010


### Comparison of General Model vs. Celtics-Specific Model Performance

#### Descriptive Statistics for Point Differences
- **General Model**
  - Mean Difference: 2.24
  - Median Difference: 1.74
  - Standard Deviation: 9.30
  - Range: (-30.10, 43.08)

- **Celtics-Specific Model**
  - Mean Difference: 12.53
  - Median Difference: 11.17
  - Standard Deviation: 23.77
  - Range: (-53.60, 62.11)

#### Hypothesis Testing: Unpaired t-test
- **t-statistic:** -3.45  
- **p-value:** 0.0010  

#### Conclusion
- The p-value (0.0010) is less than the significance level of 0.05, indicating a statistically significant difference between the two models.
- The **general model** has a much smaller mean point difference (2.24 vs. 12.53) and lower variability (std dev: 9.30 vs. 23.77) compared to the Celtics-specific model.
- This suggests that the general model performs significantly better than the Celtics-specific model on Boston Celtics data.
